src/libcore/num/dec2flt/mod.rs

```diff
@@ -232,8 +234,7 @@ fn convert<T: RawFloat>(mut decimal: Decimal) -> Result<T, ParseFloatError> {
         return Ok(x);
     }
     // AlgorithmM and AlgorithmR both compute approximately `f * 10^e`.
-    let max_digits = decimal.integral.len() + decimal.fractional.len() +
-                     decimal.exp.abs() as usize;
+    let max_digits = decimal.integral.len() + decimal.fractional.len() + decimal.exp.abs() as usize;
```

Comment:
Are you sure you're not counting the leading `+` from the diff? I'm getting 100 chars exactly without it.